src/facade/dragonfly_connection.cc

```diff
@@ -203,6 +209,13 @@ void Connection::DispatchOperations::operator()(const MonitorMessage& msg) {
   rbuilder->SendSimpleString(msg);
 }
 
+void Connection::DispatchOperations::operator()(const AclUpdateMessage& msg) {
+  auto* ctx = static_cast<dfly::ConnectionContext*>(self->cntx());
+  if (ctx && msg.username == ctx->authed_username) {
+    ctx->acl_categories = msg.categories;
+  }
```

Comment:
I first thought that you can send this message to a connection only if it already has this specific user. However if a acl setuser operation manages to execute before it it whould've been a bug. So its correct this way

Reply:
Technically the `set acluser` must complete, since it holds a write lock over the registry.